src/app/components/navbar/NavBar.jsx

```diff
@@ -13,27 +13,8 @@ const LANG = {
 };
 
 const NavBar = props => {
-    const [previewLanguage, setPreviewLanguage] = useState(null);
-
-    useEffect(() => {
-        if (cookies.get("lang")) {
-            setPreviewLanguage(cookies.get("lang"));
-        } else {
-            cookies.add("lang", "en", null);
-            setPreviewLanguage("en");
-        }
-    }, []);
-
-    useEffect(() => {
-        if (previewLanguage) {
-            changeLanguageCookie(previewLanguage);
-        }
-    }, [previewLanguage, changeLanguageCookie]);
-
-    const changeLanguageCookie = previewLanguage => {
-        cookies.remove("lang");
-        cookies.add("lang", previewLanguage, null);
-    };
+    console.log("previewLanguage", props.previewLanguage);
+    useEffect(() => props.setPreviewLanguage("en"), []);
```

Comment:
initialState.js has previewLanguage set to "en" 

Does this mean I can remove the useEffect?

Reply:
something need to transport it to here and this is what the useEffect does here. we can use useSelector if you don't like this way